### ISSUE TITLE
Error during Clear Accessory Cashe

### ISSUE THREAD

**Issue by grzegorz914** [oznu/homebridge-config-ui-x#543]
Hi @oznu 
after clear accessory cashe get error in the log:
```
}
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] RequestError: Error: socket hang up
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] Caught Unhandled Rejection Error :: Details Below
RequestError: Error: socket hang up
    at new RequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:185:22)
    at Request.emit (events.js:321:20)
    at Request.onRequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:321:20)
    at Socket.socketOnEnd (_http_client.js:463:9)
    at Socket.emit (events.js:333:22)
    at endReadableNT (_stream_readable.js:1201:12) {
  name: 'RequestError',
  message: 'Error: socket hang up',
  cause: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  error: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  options: {
    qs: { id: '1.36409,1.36410,1.36411,1.36412,1.36413' },
    json: true,
    uri: 'http://192.168.1.33:9104/characteristics',
    method: 'GET',
    callback: [Function: RP$callback],
    transform: undefined,
    simple: true,
    resolveWithFullResponse: false,
    transform2xxOnly: false
  },
  response: undefined
}
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] RequestError: Error: socket hang up
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] Caught Unhandled Rejection Error :: Details Below
RequestError: Error: socket hang up
    at new RequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:185:22)
    at Request.emit (events.js:321:20)
    at Request.onRequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:321:20)
    at Socket.socketOnEnd (_http_client.js:463:9)
    at Socket.emit (events.js:333:22)
    at endReadableNT (_stream_readable.js:1201:12) {
  name: 'RequestError',
  message: 'Error: socket hang up',
  cause: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  error: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  options: {
    qs: { id: '1.537860,1.537861,1.537862,1.537863,1.537864,1.537865' },
    json: true,
    uri: 'http://192.168.1.33:9103/characteristics',
    method: 'GET',
    callback: [Function: RP$callback],
    transform: undefined,
    simple: true,
    resolveWithFullResponse: false,
    transform2xxOnly: false
  },
  response: undefined
}
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] RequestError: Error: socket hang up
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] Caught Unhandled Rejection Error :: Details Below
RequestError: Error: socket hang up
    at new RequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:185:22)
    at Request.emit (events.js:321:20)
    at Request.onRequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:321:20)
    at Socket.socketOnEnd (_http_client.js:463:9)
    at Socket.emit (events.js:333:22)
    at endReadableNT (_stream_readable.js:1201:12) {
  name: 'RequestError',
  message: 'Error: socket hang up',
  cause: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  error: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  options: {
    qs: { id: '1.36409,1.36410,1.36411,1.36412,1.36413' },
    json: true,
    uri: 'http://192.168.1.33:9104/characteristics',
    method: 'GET',
    callback: [Function: RP$callback],
    transform: undefined,
    simple: true,
    resolveWithFullResponse: false,
    transform2xxOnly: false
  },
  response: undefined
}
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] RequestError: Error: socket hang up
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] Caught Unhandled Rejection Error :: Details Below
RequestError: Error: socket hang up
    at new RequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:185:22)
    at Request.emit (events.js:321:20)
    at Request.onRequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:321:20)
    at Socket.socketOnEnd (_http_client.js:463:9)
    at Socket.emit (events.js:333:22)
    at endReadableNT (_stream_readable.js:1201:12) {
  name: 'RequestError',
  message: 'Error: socket hang up',
  cause: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  error: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  options: {
    qs: { id: '1.36409,1.36410,1.36411,1.36412,1.36413' },
    json: true,
    uri: 'http://192.168.1.33:9104/characteristics',
    method: 'GET',
    callback: [Function: RP$callback],
    transform: undefined,
    simple: true,
    resolveWithFullResponse: false,
    transform2xxOnly: false
  },
  response: undefined
}
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] RequestError: Error: socket hang up
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] Caught Unhandled Rejection Error :: Details Below
RequestError: Error: socket hang up
    at new RequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:185:22)
    at Request.emit (events.js:321:20)
    at Request.onRequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:321:20)
    at Socket.socketOnEnd (_http_client.js:463:9)
    at Socket.emit (events.js:333:22)
    at endReadableNT (_stream_readable.js:1201:12) {
  name: 'RequestError',
  message: 'Error: socket hang up',
  cause: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  error: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  options: {
    qs: { id: '1.537860,1.537861,1.537862,1.537863,1.537864,1.537865' },
    json: true,
    uri: 'http://192.168.1.33:9103/characteristics',
    method: 'GET',
    callback: [Function: RP$callback],
    transform: undefined,
    simple: true,
    resolveWithFullResponse: false,
    transform2xxOnly: false
  },
  response: undefined
}
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] RequestError: Error: socket hang up
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] Caught Unhandled Rejection Error :: Details Below
RequestError: Error: socket hang up
    at new RequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:185:22)
    at Request.emit (events.js:321:20)
    at Request.onRequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:321:20)
    at Socket.socketOnEnd (_http_client.js:463:9)
    at Socket.emit (events.js:333:22)
    at endReadableNT (_stream_readable.js:1201:12) {
  name: 'RequestError',
  message: 'Error: socket hang up',
  cause: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  error: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  options: {
    qs: { id: '1.36409,1.36410,1.36411,1.36412,1.36413' },
    json: true,
    uri: 'http://192.168.1.33:9104/characteristics',
    method: 'GET',
    callback: [Function: RP$callback],
    transform: undefined,
    simple: true,
    resolveWithFullResponse: false,
    transform2xxOnly: false
  },
  response: undefined
}
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] RequestError: Error: socket hang up
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] Caught Unhandled Rejection Error :: Details Below
RequestError: Error: socket hang up
    at new RequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:185:22)
    at Request.emit (events.js:321:20)
    at Request.onRequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:321:20)
    at Socket.socketOnEnd (_http_client.js:463:9)
    at Socket.emit (events.js:333:22)
    at endReadableNT (_stream_readable.js:1201:12) {
  name: 'RequestError',
  message: 'Error: socket hang up',
  cause: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  error: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  options: {
    qs: { id: '1.36409,1.36410,1.36411,1.36412,1.36413' },
    json: true,
    uri: 'http://192.168.1.33:9104/characteristics',
    method: 'GET',
    callback: [Function: RP$callback],
    transform: undefined,
    simple: true,
    resolveWithFullResponse: false,
    transform2xxOnly: false
  },
  response: undefined
}
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] RequestError: Error: socket hang up
[2/15/2020, 8:31:56 AM] [homebridge-config-ui-x] Caught Unhandled Rejection Error :: Details Below
RequestError: Error: socket hang up
    at new RequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:185:22)
    at Request.emit (events.js:321:20)
    at Request.onRequestError (/usr/lib/node_modules/homebridge-config-ui-x/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:321:20)
    at Socket.socketOnEnd (_http_client.js:463:9)
    at Socket.emit (events.js:333:22)
    at endReadableNT (_stream_readable.js:1201:12) {
  name: 'RequestError',
  message: 'Error: socket hang up',
  cause: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  error: Error: socket hang up
      at connResetException (internal/errors.js:609:14)
      at Socket.socketOnEnd (_http_client.js:463:23)
      at Socket.emit (events.js:333:22)
      at endReadableNT (_stream_readable.js:1201:12)
      at processTicksAndRejections (internal/process/task_queues.js:84:21) {
    code: 'ECONNRESET'
  },
  options: {
    qs: { id: '1.20911' },
    json: true,
    uri: 'http://192.168.1.33:9107/characteristics',
    method: 'GET',
    callback: [Function: RP$callback],
    transform: undefined,
    simple: true,
    resolveWithFullResponse: false,
    transform2xxOnly: false
  },
  response: undefined
}
```

**Comment by oznu** [comment]
I'm not sure about this one sorry. Is it still an issue after restarting everything?

**Comment by grzegorz914** [comment]
Yes, The issue still but I think this comming from openhab2-complete plugin.

**Comment by oznu** [comment]
Ok I think I figured out how/why this happens. It's because when clearing the accessory cache I'm only restarting the Homebridge service, not the UI service - this is a "new" way of doing things as every other restart operation before this would have just restarted the entire UI at the same time.

I've updated the `hap-client` dependency to catch these errors now so it shouldn't crash the UI process when the connections for accessory control disconnect.

Currently in `@test`.

Same issue: https://github.com/oznu/homebridge-config-ui-x/issues/583
